### PR TITLE
Feature/filter open

### PR DESCRIPTION
**Pull request by amitschendel**

### CODE REVIEW

File: pkg/collector/collector.go [hunk 1]
```diff
@@ -733,3 +766,58 @@ func openEventExists(openEvent *tracing.OpenEvent, openEvents []OpenCalls) (bool
...
 }
+
+func (cm *CollectorManager) shouldIncludeOpenEvent(openEvent *tracing.OpenEvent, openEvents []OpenCalls) bool {
+	// Check if we exceeded the maximum number of open events.
+	if len(openEvents) > MaxOpenEvents {
+		return false
+	}
+
+	// Check if we should ignore this path.
+	if os.Getenv("OPEN_IGNORE_PREFIXES") != "" {
```
Comment (slashben) [inline pkg/collector/collector.go:777]: The best practice to do these configs in main, not in a nested package

File: pkg/collector/collector.go [hunk 2]
```diff
@@ -733,3 +766,58 @@ func openEventExists(openEvent *tracing.OpenEvent, openEvents []OpenCalls) (bool
...
+	hasSamePath, hasSameFlags := openEventExists(openEvent, openEvents)
+	if hasSamePath && hasSameFlags {
+		return false
+	}
+
+	// Check if we should ignore mounts.
+	if os.Getenv("OPEN_IGNORE_MOUNTS") == "true" {
+		cm.podMountCacheMutex.Lock()
+		mounts := cm.podMountCache[fmt.Sprintf("%s-%s", openEvent.PodName, openEvent.Namespace)]
+		cm.podMountCacheMutex.Unlock()
```
Comment (slashben) [inline pkg/collector/collector.go:796]: maybe move this to the caller function to prevent too many lock

**Review by slashben — APPROVED** [review]

File: pkg/collector/collector.go (hunk 1 above)
Comment (amitschendel) [reply pkg/collector/collector.go]: Done

File: pkg/collector/collector.go (hunk 2 above)
Comment (amitschendel) [reply pkg/collector/collector.go]: Done

**Review by slashben — APPROVED** [review]

**Pull request merged**